### PR TITLE
Output ReHo as a CIFTI

### LINKED ISSUE #260: [ENH] Combine surface rehos and subcortical reho to scalar/dtseries
- [x] Check if this has been implemented

### PR DESCRIPTION
**Pull request by tsalo**
Closes #260.

The breaking change is that, instead of outputting two surface ReHo files (`<source_entities>_space-fsLR_den-32k_hemi-L_reho.func.gii` and `<source_entities>_space-fsLR_den-32k_hemi-R_reho.func.gii`), the cifti workflow will now output a single cifti (`<source_entities>_space-fsLR_den-91k_reho.dscalar.nii`).

## Changes proposed in this pull request
- Add interfaces for `wb_command  -cifti-separate * -volume-all * -label *` (`CiftiSeparateVolumeAll`) and `wb_command -cifti-create-dense-scalar` (`CiftiCreateDenseScalar`).
    - We _could_ create a general `wb_command -cifti-separate` interface that does the work of `CiftiSeparateMetric` and `CiftiSeparateVolumeAll`, but that command is pretty complicated and xcpd only uses it in a couple of ways, so it may not be worth it.
- Make mask optional in `ReHoNamePatch`.
- Rename `init_surface_reho_wf` to `init_cifti_reho_wf`, since it now outputs a cifti file and not just the surface giftis.
- Rename `init_3d_reho_wf` to `init_nifti_reho_wf` (to match `init_cifti_reho_wf`).
- Refactor `init_cifti_reho_wf` to output the subcortical volumetric data from the cleaned BOLD cifti into a nifti, then calculate ReHo on that nifti, and then recombine that nifti and the two surface ReHo files into a single cifti.

## Documentation that should be reviewed
- The outputs list.

### CODE REVIEW

File: xcp_d/interfaces/workbench.py [hunk 1]
```diff
@@ -816,3 +816,139 @@ class CiftiSeparateMetric(WBCommand):
     input_spec = _CiftiSeparateMetricInputSpec
     output_spec = _CiftiSeparateMetricOutputSpec
```
Comment (kahinimehta) [inline xcp_d/interfaces/workbench.py:817]: do we still need the cifti separate metric?

**Review by kahinimehta — APPROVED** [review]

File: xcp_d/interfaces/workbench.py (hunk 1 above)
Comment (tsalo) [reply xcp_d/interfaces/workbench.py:817]: We could write one interface that covers all of `-cifti-separate` (including both `-metric` to get the surfaces and `-volume-all` to get the subcortical data), but I was feeling lazy and went with a separate interface.

File: xcp_d/workflow/restingstate.py [hunk 2]
```diff
@@ -209,39 +216,76 @@ def init_surface_reho_wf(
...
+    # Merge the surfaces and subcortical structures back into a CIFTI
+    merge_cifti = pe.Node(
+        CiftiCreateDenseScalar(),
+        name="merge_cifti",
+        mem_gb=mem_gb,
+        n_procs=omp_nthreads,
+    )
+
     # Write out results
     workflow.connect([
```
Comment (kahinimehta) [inline xcp_d/workflow/restingstate.py:277]: I think the parentheses might be off?

Comment (tsalo) [reply xcp_d/workflow/restingstate.py:277]: You're right! Thanks!

File: xcp_d/workflow/restingstate.py [hunk 3]
```diff
@@ -209,39 +216,76 @@ def init_surface_reho_wf(
...
         (inputnode, lh_surf, [('clean_bold', 'in_file')]),
         (inputnode, rh_surf, [('clean_bold', 'in_file')]),
+        (inputnode, subcortical_nifti, [('clean_bold', 'in_file')]),
         (lh_surf, lh_reho, [('out_file', 'surf_bold')]),
         (rh_surf, rh_reho, [('out_file', 'surf_bold')]),
-        (lh_reho, outputnode, [('surf_gii', 'lh_reho')]),
-        (rh_reho, outputnode, [('surf_gii', 'rh_reho')]),
+        (subcortical_nifti, subcortical_reho, [('out_file', 'in_file')]),
+        (subcortical_nifti, merge_cifti, ['label_file', 'structure_label_volume']),
```
Comment (kahinimehta) [inline xcp_d/workflow/restingstate.py]: missing ()

**Comment by kahinimehta** [comment]
Might also have to update the pytests

**Review by kahinimehta — APPROVED** [review]

**Comment by tsalo** [comment]
Merging now!

**Pull request merged**